agents/postgres/pgstatmonitor/pgstatmonitor.go

```diff
@@ -307,12 +242,12 @@ func (m *PGStatMonitorQAN) Run(ctx context.Context) {
 				m.changes <- agents.Change{Status: inventorypb.AgentStatus_STARTING}
 			}
 
-			lengthS := uint32(m.waitTime.Seconds())
+			lengthS := uint32(waitTime.Seconds())
 			buckets, err := m.getNewBuckets(ctx, lengthS)
```

Comment:
I think you can call settings here and pass them to the `getNewBuckets` method, in that case, you will be able to support immediate change for `waitTime` as well.

Reply:
Done.